WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2107,11 +2107,12 @@ public void run() {
                 } else {
                     // discard post if new & empty
                     if (isDiscardable()) {
-                        mPostEditorAnalyticsSession.setOutcome(Outcome.CANCEL);
                         mDispatcher.dispatch(PostActionBuilder.newRemovePostAction(mPost));
                     }
                     removePostOpenInEditorStickyEvent();
                     if (doFinish) {
+                        // if we shouldn't save and we should exit, set the session tracking outcome to CANCEL
+                        mPostEditorAnalyticsSession.setOutcome(Outcome.CANCEL);
```

Comment:
also moved this line so we properly get a CANCEL outcome when the user opens an existing non-empty post and then does nothing and exits: we were saying "SAVE" but it's actually a CANCEL as per the specs in https://github.com/wordpress-mobile/gutenberg-mobile/issues/556